pkg/healthcheck/healthcheck_output.go

```diff
@@ -166,98 +168,117 @@ func RunExtensionsChecks(wout io.Writer, werr io.Writer, extensions []string, fl
+func runChecksShort(wout io.Writer, hc Runner) bool {
 	var lastCategory CategoryID
-	var prettyPrintResults CheckObserver
 
 	spin := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
 	spin.Writer = wout
 
-	if short {
-		prettyPrintResults = func(result *CheckResult) {
-			spin.Stop()
-			if result.Retry {
-				if isatty.IsTerminal(os.Stdout.Fd()) {
-					spin.Suffix = fmt.Sprintf(" %s", result.Err)
-					spin.Color("bold") // this calls spin.Restart()
-				}
-				return
+	prettyPrintResults := func(result *CheckResult) {
+		spin.Stop()
+		if result.Retry {
+			if isatty.IsTerminal(os.Stdout.Fd()) {
+				spin.Suffix = fmt.Sprintf(" %s", result.Err)
+				spin.Color("bold") // this calls spin.Restart()
 			}
+			return
+		}
 
-			status := okStatus
-			if result.Err != nil {
-				status = failStatus
-				if result.Warning {
-					status = warnStatus
-				}
-			}
+		if result.Err == nil {
+			return
+		}
 
-			if status == okStatus {
-				return
-			}
+		status := failStatus
+		if result.Warning {
+			status = warnStatus
+		}
 
-			if lastCategory != result.Category {
-				if lastCategory != "" {
-					fmt.Fprintln(wout)
-				}
+		if lastCategory != result.Category {
+			if lastCategory != "" {
+				fmt.Fprintln(wout)
+			}
 
-				fmt.Fprintln(wout, result.Category)
-				fmt.Fprintln(wout, strings.Repeat("-", len(result.Category)))
+			fmt.Fprintln(wout, result.Category)
+			fmt.Fprintln(wout, strings.Repeat("-", len(result.Category)))
 
-				lastCategory = result.Category
-			}
+			lastCategory = result.Category
+		}
 
-			fmt.Fprintf(wout, "%s %s\n", status, result.Description)
-			if result.Err != nil {
-				fmt.Fprintf(wout, "    %s\n", result.Err)
-				if result.HintURL != "" {
-					fmt.Fprintf(wout, "    see %s for hints\n", result.HintURL)
-				}
+		fmt.Fprintf(wout, "%s %s\n", status, result.Description)
+		if result.Err != nil {
+			fmt.Fprintf(wout, "    %s\n", result.Err)
+			if result.HintURL != "" {
+				fmt.Fprintf(wout, "    see %s for hints\n", result.HintURL)
 			}
 		}
+	}
+
+	success := hc.RunChecks(prettyPrintResults)
+	// this empty line separates final results from the checks list in the output
+	fmt.Fprintln(wout, "")
+
+	if !success {
+		fmt.Fprintf(wout, "Status check results are %s\n", failStatus)
 	} else {
-		prettyPrintResults = func(result *CheckResult) {
-			if lastCategory != result.Category {
-				if lastCategory != "" {
-					fmt.Fprintln(wout)
-				}
+		fmt.Fprintf(wout, "Status check results are %s\n", okStatus)
+	}
+
+	return success
+}
 
-				fmt.Fprintln(wout, result.Category)
-				fmt.Fprintln(wout, strings.Repeat("-", len(result.Category)))
+func runChecksTable(wout io.Writer, hc Runner) bool {
+	var lastCategory CategoryID
+	var prettyPrintResults CheckObserver
 
-				lastCategory = result.Category
+	spin := spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+	spin.Writer = wout
+
+	prettyPrintResults = func(result *CheckResult) {
```

Comment:
The `var prettyPrintResults CheckObserver` can be removed:
```suggestion
	prettyPrintResults := func(result *CheckResult) {
```